tools/autograd/templates/Functions.cpp

```diff
@@ -1792,6 +1792,55 @@ Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const T
   return u_term + sigma_term + v_term;
 }
 
+Tensor eig_backward(const std::vector<torch::autograd::Variable> &grads, const Tensor& self,
+                    bool eigenvectors, const Tensor& lambda, const Tensor& v) {
+  // This gradient only works for real eigenvalues at the moment.
+  TORCH_CHECK(eigenvectors,
+           "eig_backward: Setting eigenvectors to false in torch.eig doesn't compute eigenvectors ",
+           "and hence we cannot compute backward. Please use torch.eig(eigenvectors=True)");
+  auto imag_idx = at::ones({1}, lambda.options().dtype(at::kLong));
+  auto real_idx = at::zeros({1}, lambda.options().dtype(at::kLong));
+  auto zeros = at::zeros({1}, lambda.options());
+  TORCH_CHECK(
+      at::allclose(
+        at::index_select(lambda, /*dim=*/-1, /*index=*/imag_idx),
```

Comment:
If this is only selecting the `1`th entry in the last dimension, a more efficient way to do it is with `lambda.index(-1, 1)`.
`index_select()` copies the values while `index()` does not.

Reply:
`.index()` seems to be accepting list of tensor index and the documentation does not seem to be clear, so I use `.slice()`. Hopefully `slice()` does not make a copy.